src/mkdocs_bibtex/utils.py

```diff
@@ -168,3 +169,21 @@ def format_bibliography(citation_quads):
         bibliography.append(bibliography_text)
 
     return "\n".join(bibliography)
+
+
+def tempfile_from_url(url, suffix='.bib'):
+    attempts = 3
+    while attempts > 0:
+        try:
+            dl = requests.get(url)
+            if dl.status_code != 200:
+                raise Exception(f"Status Code: {dl.status_code}")
+            file = tempfile.NamedTemporaryFile(mode='w', suffix=suffix, delete=False)
+            file.write(dl.text)
+            file.close()
+            return file.name
+        except Exception as e:
+            attempts -= 1
+            if attempts == 0:
+                raise Exception(f"Encountered issues while trying to download {url}, {e}")
+            continue
```

Comment:
Doing a few things here:
1. Switching to a for loop since the intent is to run a fixed amount of times. Having code that looks the way you wanted it to run helps people debug.
2. Using `RequestException` to ensure that the try except only catches the relevant exception
3. Raise `RuntimeError` so that we flag this is as a "failed to execute" problem.


```suggestion
    for i in range(3):
        try:
            dl = requests.get(url)
            if dl.status_code != 200:
                raise RuntimeError(f"Couldn't download the url: {url}.\n Status Code: {dl.status_code}")

            file = tempfile.NamedTemporaryFile(mode='w', suffix=suffix, delete=False)
            file.write(dl.text)
            file.close()
            return file.name
            
        except RequestException as e:
        	pass
	raise RuntimeError(f"Couldn't successfully download the url: {url}")
```